programs/axelar-solana-its/src/processor/mod.rs

```diff
@@ -247,7 +265,7 @@ pub fn process_instruction<'a>(
             data,
             gas_value,
             signing_pda_bump,
-        } => interchain_transfer::process_outbound_transfer(
+        } => interchain_transfer::process_user_interchain_transfer(
```

Comment:
This annoys me: 

- I see no reason why `CallContractWithInterchainToken` might not also be called by programs.
- I split the `InterchainTransfer` instruction, introducing  `ProgramInterchainTransfer` to avoid adding complexity in the original instruction
- to stay consistent I may have to add a `ProgramCallContractWithInterchainToken`, and I feel like the now solution is worse than the original problem

Reply:
Yeah, it's not great, but I would go with adding `ProgramCallContractWithInterchainToken`. Also, on another subject, we could replace the `Program` prefix with the `Cpi` suffix, like the mpl metadata program [does](https://developers.metaplex.com/guides/rust/how-to-cpi-into-a-metaplex-program) as that might help users (devs) that are already familiar with Solana.